From 9f4d89fe7bc5f1a73ffeadff01ee6c52ed32f1fe Mon Sep 17 00:00:00 2001
From: Guojing Yuan <guojing@google.com>
Date: Tue, 28 Jan 2025 11:57:16 -0800
Subject: [PATCH] Checks NLS component name instead of package name

Bug: 386216637
Flag: EXEMPT bugfix
Test: CTS
(cherry picked from commit f5ed187ffb6f5a25ac1c06a1e8b023b0c7d8d4b8)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:19c9b3d4e30ad6477103c46b3cdb2732215f8a12)
Merged-In: I1be52fcb3d52b8b84353ea00fd57d8360ddd966c
Change-Id: I1be52fcb3d52b8b84353ea00fd57d8360ddd966c
---
 .../notification/NotificationAccessConfirmationActivity.java  | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/com/android/settings/notification/NotificationAccessConfirmationActivity.java b/src/com/android/settings/notification/NotificationAccessConfirmationActivity.java
index 1adeb644155..7d3e7a0d698 100644
--- a/src/com/android/settings/notification/NotificationAccessConfirmationActivity.java
+++ b/src/com/android/settings/notification/NotificationAccessConfirmationActivity.java
@@ -124,7 +124,7 @@ public class NotificationAccessConfirmationActivity extends Activity
                 NLSIntent, /* flags */ 0, mUserId);
         boolean hasNLSIntentFilter = false;
         for (ResolveInfo service : matchedServiceList) {
-            if (service.serviceInfo.packageName.equals(mComponentName.getPackageName())) {
+            if (service.serviceInfo.getComponentName().equals(mComponentName)) {
                 if (!requiredPermission.equals(service.serviceInfo.permission)) {
                     Slog.e(LOG_TAG, "Service " + mComponentName + " lacks permission "
                             + requiredPermission);
@@ -158,7 +158,7 @@ public class NotificationAccessConfirmationActivity extends Activity
                 .installContent(p);
         // Consistent with the permission dialog
         // Used instead of p.mCancelable as that is only honored for AlertDialog
-        getWindow().setCloseOnTouchOutside(false); 
+        getWindow().setCloseOnTouchOutside(false);
     }
 
     @Override
-- 
2.34.1

